### PR TITLE
Manifest fix

### PR DESCRIPTION
**Pull request by hytonhan**
Fixed manager crashing on startup if bad manifests existed

### CODE REVIEW

File: ModManager/MapSystem/MapManifestFinder.cs [hunk 1]
```diff
@@ -23,14 +24,23 @@ public IEnumerable<Manifest> Find()
...
-            return manifests;
+            try
+            {
+                List<MapManifest> manifests = _persistenceService.LoadObject<List<MapManifest>>(manifestPath, false);
+                UpdateManifestInfo(manifests);
+                return manifests;
+            }
+            catch (JsonSerializationException ex)
+            {
+                return new List<MapManifest>();
```
Comment (KYPremco) [inline ModManager/MapSystem/MapManifestFinder.cs:38]: Add a log here with just the message, maybe some info text how to fix it

File: ModManager/ModSystem/ModManifestFinder.cs [hunk 2]
```diff
@@ -47,13 +64,14 @@ private Manifest LoadManifest(string manifestPath)
...
-
                 manifest.Enabled = !Path.GetExtension(manifestPath).Equals(Names.Extensions.Disabled);
-
                 manifest.RootPath = Path.GetDirectoryName(manifestPath)!;
-
                 return manifest;
             }
+            catch (JsonSerializationException ex)
+            {
+                return null;
```
Comment (KYPremco) [inline ModManager/ModSystem/ModManifestFinder.cs:73]: Also log :)

**Review by KYPremco — CHANGES_REQUESTED** [review]
Pretty much good to go

**Review by KYPremco — APPROVED** [review]

File: ModManager/MapSystem/MapManifestFinder.cs (hunk 1 above)
Comment (hytonhan) [reply ModManager/MapSystem/MapManifestFinder.cs:44]: Added

File: ModManager/ModSystem/ModManifestFinder.cs (hunk 2 above)
Comment (hytonhan) [reply ModManager/ModSystem/ModManifestFinder.cs:76]: logging added

**Pull request merged**